pkg/minikube/constants/constants.go

```diff
@@ -34,7 +34,9 @@ const (
 	// OldestKubernetesVersion is the oldest Kubernetes version to test against
 	OldestKubernetesVersion = "v1.11.10"
 	// DefaultMachineName is the default name for the VM
-	DefaultMachineName = "minikube"
+	DefaultMachineName = "minikube-minikube"
```

Comment:
I don't think DefaultMachineName should exist, but if it must - can we pick a name that isn't so repetitive? 

I feel like may just cause problems, and that machine names should be dynamically generated with something like  `minikube-<profile>-<index>`. I realize that this is effectively repetitive given our current default, but I also think we should consider renaming our default profile to `default` to reduce repetition at some point as well.

Reply:
I suspect we can just get rid of it. The MachineName func should be able to be invoked anywhere we currently use this.

Reply:
and yes, we are planning the machine name to be exactly of that format, I just haven't made the change on this PR yet,

Reply:
Just to follow-up: I think DefaultMachineName and DefaultNodeName should be removed. I think you are leaning toward this already, but my gut instinct is that:

* Seen from the VM, nodes should be named something simple like `minikube01`, `mini01` or `m01`. 
* Seen from the  hypervisor, nodes should be named `<cluster>-minikube01`, `<cluster>-mini01`, or `<cluster>-m01`, or `<cluster>-01`. 

Probably the least confusing is a scenario where the node sees themself as `m01`, and the hypervisor sees nodes as `<cluster>-m01`. 



Nodes should just be named <cluster>-<index>

Reply:
This definitely makes the most sense. I had picked the VM name and node name being the same for simplicity, but this would be much clearer.